### PR TITLE
Added layers example

### PR DESCRIPTION
**Pull request by aardgoose**
There are no examples demonstrating layers apart from some use in some of the VR demos which don't really highlight their utility (layers seen to be overlooked).

Also added a flat background texture which also has no examples.

### CODE REVIEW

File: examples/webgl_layers.html [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+				info.style.width = '100%';
+				info.style.textAlign = 'center';
+				info.innerHTML = '<a href="http://threejs.org" target="_blank" rel="noopener">three.js</a> webgl - layers';
+				container.appendChild( info );
+
+				var bgTexture = new THREE.TextureLoader().load( 'textures/brick_diffuse.jpg' );
+				var bgColor = new THREE.Color( 0xf0f0f0 );
+
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 10000 );
+				camera.layers.enable( 0 );
```
Comment (Mugen87) [inline examples/webgl_layers.html]: Maybe you could add a comment that says channel zero is enabled by default?

File: examples/webgl_layers.html [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				var info = document.createElement( 'div' );
+				info.style.position = 'absolute';
+				info.style.top = '10px';
+				info.style.width = '100%';
+				info.style.textAlign = 'center';
+				info.innerHTML = '<a href="http://threejs.org" target="_blank" rel="noopener">three.js</a> webgl - layers';
+				container.appendChild( info );
```
Comment (Mugen87) [inline examples/webgl_layers.html]: Please write this code as HTML markup.

File: examples/webgl_layers.html [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+				var layers = { red: true, green: true, blue: true, background: false };
+
+				//
+				// Init gui
+				var gui = new dat.GUI();
+				gui.add( layers, 'red' ).onChange( function () { camera.layers.toggle( 0 ); } );
+				gui.add( layers, 'green' ).onChange( function () { camera.layers.toggle( 1 ); } );
+				gui.add( layers, 'blue' ).onChange( function () { camera.layers.toggle( 2 ); } );
+
+				gui.add( layers, 'background' ).onChange( function ( value ) { scene.background = value ? bgTexture : bgColor; } );
```
Comment (Mugen87) [inline examples/webgl_layers.html]: I would remove the background since it feels out of place in this example. Let's just focus on layers.

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: examples/webgl_layers.html (hunk 2 above)
Comment (Mugen87) [reply examples/webgl_layers.html]: Maybe this one:
https://github.com/mrdoob/three.js/blob/master/examples/webgl_loader_fbx.html

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
For testing: https://rawgit.com/aardgoose/three.js/layers-example/examples/webgl_layers.html

File: examples/webgl_layers.html [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+			function render() {
+
+				theta += 0.1;
+
+				camera.position.x = radius * Math.sin( THREE.Math.degToRad( theta ) );
+				camera.position.y = radius * Math.sin( THREE.Math.degToRad( theta ) );
+				camera.position.z = radius * Math.cos( THREE.Math.degToRad( theta ) );
+				camera.lookAt( scene.position );
+
+				camera.updateMatrixWorld();
```
Comment (WestLangley) [inline examples/webgl_layers.html]: Can you explain why are you calling `camera.updateMatrixWorld()` here?

File: examples/webgl_layers.html [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 10000 );
+				camera.layers.enable( 0 ); // enabled by default
+				camera.layers.enable( 1 );
+				camera.layers.enable( 2 );
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0xf0f0f0 );
+
+				var light = new THREE.DirectionalLight( 0xffffff, 1 );
```
Comment (WestLangley) [inline examples/webgl_layers.html]: How about
```javascript
var light = new THREE.PointLight( 0xffffff, 1 );
scene.add( camera );
camera add( light );
```

**Comment by WestLangley** [comment]
>Also added a flat background texture which also has no examples.

There is no texture, but I don't think it is necessary in this example.

**Comment by aardgoose** [comment]
Yes, I removed it from the first version,

File: examples/webgl_layers.html (hunk 4 above)
Comment (aardgoose) [reply examples/webgl_layers.html]: cut and paste from another example - I'll review it.

File: examples/webgl_layers.html (hunk 5 above)
Comment (aardgoose) [reply examples/webgl_layers.html]: lighting fixed as you suggested.

File: examples/webgl_layers.html [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 1, 10000 );
+				camera.layers.enable( 0 ); // enabled by default
+				camera.layers.enable( 1 );
+				camera.layers.enable( 2 );
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0xf0f0f0 );
+
+				var light = new THREE.PointLight( 0xffffff, 1 );
+				light.position.set( 1, 1, 1 ).normalize();
```
Comment (WestLangley) [inline examples/webgl_layers.html]: Why are you normalizing the position?

Why isn't ( 0, 0, 0 ), the default, adequate?

Comment (aardgoose) [reply examples/webgl_layers.html]: junk left in by mistake. apologies.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**